internal/pkg/overlay/overlay.go

```diff
@@ -352,6 +352,18 @@ func RenderTemplateFile(fileName string, data TemplateStruct) (
 	err error) {
 	backupFile = true
 	writeFile = true
+	// parse the overlay name out of the absolute path
+	overlayPath, _ := filepath.Rel(OverlaySourceTopDir(), fileName)
+	withoutRootfs, err := filepath.Rel("rootfs", overlayPath)
+	if err != nil {
+		overlayPath = withoutRootfs
+	}
+	overlayPathElem := strings.Split(overlayPath, "/")
+	if len(overlayPathElem) > 1 {
+		data.Overlay = overlayPathElem[0]
+	} else {
+		data.Overlay = "UNDEF"
```

Comment:
I think an empty string would be more appropriate here, rather than hard-coding a string value "UNDEF". Any objection?

Reply:
An empty string avoids the problem of an overlay named `UNDEF`. I can't think of a case where I'd want this to return anything other than an empty string if I called it outside the context of an overlay.